### ISSUE TITLE
no-useless-cast with Map.get

### ISSUE THREAD

**Issue by AntonShchemelev** [SonarSource/SonarTS#839]
**I want to report a bug.**

**Node.js version:** 11.11.0<br>
**TypeScript version:** 3.3.3333<br>
**tslint-sonarts version and used IDE:** 1.9.0 Visual Studio Code latest<br>
**TSLint version:** 5.14.0<br>
**Rule key:** no-useless-cast

**Reproducer** 
```JS
const m = new Map<string, string>();
m.set("x", "y");
const y = m.get("y")!.length; // False-positive here
console.log(y);
```

**Logs**
```
Remove this unnecessary not-null assertion. (no-useless-cast)tslint(1)
```

**Expected behavior**

`tslint-sonarts` should not give the warning because if I remove not null assertion operator `!` the warning is gone but TypeScript fails to compile the code with the error:

```
error TS2532: Object is possibly 'undefined'.
const y = m.get("y").length;
```
